engine/access/rpc/backend/backend_block_details.go

```diff
@@ -58,3 +58,15 @@ func (b *backendBlockDetails) GetBlockByHeight(_ context.Context, height uint64)
 
 	return block, nil
 }
+
+func (b *backendBlockDetails) GetBlockStatus(_ context.Context, id flow.Identifier) flow.BlockStatus {
+	block, err := b.blocks.ByID(id)
+	if err != nil {
+		return flow.BlockStatusUnknown
+	}
+	_, err = b.blocks.ByHeight(block.Header.Height + 1)
```

Comment:
I don't think this is an accurate way to get the status. Instead, you can use `b.state.Sealed().Head()` to get the latest sealed header and compare heights.